common.zig

```diff
@@ -26,19 +26,28 @@ pub const ConnectResult = struct {
 pub fn connect(allocator: std.mem.Allocator) !ConnectResult {
     const display = x.getDisplay();
 
-    const sock = x.connect(display) catch |err| {
+    var connection_buf: [1024]u8 = undefined;
+    var connection_fixed_buf = std.heap.FixedBufferAllocator.init(&connection_buf);
+    var connection = x.connect(connection_fixed_buf.allocator(), display) catch |err| {
         std.log.err("failed to connect to display '{s}': {s}", .{display, @errorName(err)});
         std.os.exit(0xff);
     };
+    defer connection.deinit();
 
     {
-        const len = comptime x.connect_setup.getLen(0, 0);
-        var msg: [len]u8 = undefined;
-        x.connect_setup.serialize(&msg, 11, 0, .{ .ptr = undefined, .len = 0 }, .{ .ptr = undefined, .len = 0 });
-        try send(sock, &msg);
+        const auth_name = connection.authentication.getName();
+        const auth_data = connection.authentication.getData();
+
+        const len = x.connect_setup.getLen(@intCast(auth_name.len), @intCast(auth_data.len));
+        var msg_buf: [1024]u8 = undefined;
```

Comment:
@marler8997 I remember you saying that there is a known max length for this. Do you remember what it is?

Reply:
We could add these constants to the `connect_setup` struct namespace:
```zig
pub const connect_setup = struct {
    pub const max_auth_proto_name_len = std.math.maxInt(u16);
    pub const max_auth_proto_data_len = std.math.maxInt(u16);
    pub const max_len = getLen(max_auth_proto_name_len, max_auth_proto_data_len);
};
```

Reply:
Oh ok I thought there was a defined max limit that was more reasonable. 128kb for this fields seems overkill (it will realistically never even be more than 128 bytes), so this 1kb limit is probably more reasonable.